apps/rush-lib/src/cli/actions/ListAction.ts

```diff
@@ -1,76 +1,98 @@
+      parameterShortName: "-v",
+      description:
+        "If this flag is specified, the project version will be " +
+        "displayed in a column along with the package name."
     });
 
     this._path = this.defineFlagParameter({
-      parameterLongName: '--path',
-      parameterShortName: '-p',
-      description: 'If this flag is specified, the project path will be ' +
-        'displayed in a column along with the package name.'
+      parameterLongName: "--path",
+      parameterShortName: "-p",
+      description:
+        "If this flag is specified, the project path will be " +
+        "displayed in a column along with the package name."
     });
 
     this._fullPath = this.defineFlagParameter({
-      parameterLongName: '--full-path',
-      parameterShortName: '-f',
-      description: 'If this flag is specified, the project full path will ' +
-        'be displayed in a column along with the package name.'
+      parameterLongName: "--full-path",
+      parameterShortName: "-f",
+      description:
+        "If this flag is specified, the project full path will " +
+        "be displayed in a column along with the package name."
+    });
+
+    this._jsonFlag = this.defineFlagParameter({
+      parameterLongName: "--json",
+      description: "If specified,  a json file of result will be generated."
     });
   }
 
   protected run(): Promise<void> {
     return Promise.resolve().then(() => {
-      const allPackages: Map<string, RushConfigurationProject> = this.rushConfiguration.projectsByName;
+      const allPackages: Map<string, RushConfigurationProject> = this
+        .rushConfiguration.projectsByName;
       if (this._version.value || this._path.value || this._fullPath.value) {
         this._printListTable(allPackages);
+      } else if (this._jsonFlag.value) {
+        this._serializeToJson(allPackages);
       } else {
         this._printList(allPackages);
       }
     });
   }
 
+  private _serializeToJson(
+    allPackages: Map<string, RushConfigurationProject>
+  ): void {
+    allPackages.forEach((_config: RushConfigurationProject, name: string) => {
+      console.log(JSON.stringify(name));
+    });
+  }
+
   private _printList(allPackages: Map<string, RushConfigurationProject>): void {
     allPackages.forEach((_config: RushConfigurationProject, name: string) => {
       console.log(name);
     });
   }
 
-  private _printListTable(allPackages: Map<string, RushConfigurationProject>): void {
-    const tableHeader: string[] = ['Project'];
+  private _printListTable(
+    allPackages: Map<string, RushConfigurationProject>
+  ): void {
+    const tableHeader: string[] = ["Project"];
     if (this._version.value) {
-      tableHeader.push('Version');
+      tableHeader.push("Version");
     }
     if (this._path.value) {
-      tableHeader.push('Path');
+      tableHeader.push("Path");
     }
     if (this._fullPath.value) {
-      tableHeader.push('Full Path');
+      tableHeader.push("Full Path");
```

Comment:
Revert this as well.

Are you using Prettier or something?

Reply:
Yeah I use prettier in VS code on my own laptop.. Will discard it and revert changes back